### PR TITLE
Add WASM compilation and split workspace for web and native

### PR DESCRIPTION
**Pull request by mycognosist**
While there appears to be _a lot_ going on in this PR, the changes are fairly straightforward:

- The existing validation code (using `node_bindgen`) has been moved into `native/` (Node.js)
- Equivalent code for generating a WebAssembly module (using `wasm_bindgen`) has been added in `web/` (WASM)
  - Included for now are various assets which facilitate development and testing:
    - `web/server.py` is a web server which serves assets with the necessary `COEP` and `COOP` headers (required for [cross-origin isolation](https://web.dev/coop-coep/))
    - `web/rust-toolchain` and `web/.cargo/config` provide compilation configuration for WASM
    - `web/index.html` is a simple web page which loads the WASM module
    - `web/index.js` provides a JS wrapper around the WASM functions (mirrors the API of `native/index.js`)
    - `web/wasmLoader.js` is a simple module used to call and test the validation functions
- [Cargo workspaces](https://doc.rust-lang.org/book/ch14-03-cargo-workspaces.html) have been introduced to split the package into multiple libraries (one for `native` and one for `web`)
- The root `README.md` has been changed to focus exclusively on usage and behaviour (user-oriented)
  - The `native` and `web` directories each host a separate `README.md` with implementation-specific details

### CODE REVIEW

**Comment by staltz** [comment]
Looking good so far!

**Comment by staltz** [comment]
Thoughts and suggestions:

- Add the python file (and other files, like the HTML) to `.npmignore` so that it's not publish to the npm package
- Make a file `api.js` that takes the duplicate code in `native/index.js` and `web/index.js`, then `native/index.js` and `web/index.js` import the utils in `api.js`

**Comment by mycognosist** [comment]
I'm keen to merge this PR before I get carried away with writing tests for the WASM (would rather do that in a separate PR so it's more manageable).

I'm not entirely sure how to deduplicate the `index.js` code into a single `api.js` (the conditional import part is what's confusing me) so I've left that out for now and we can address it during a pairing session.

File: web/.npmignore [hunk 1]
```diff
@@ -0,0 +1,5 @@
+index.html
+server.py
+wasmLoader.js
+rust-toolchain
+.cargo/config
```
Comment (staltz) [inline web/.npmignore:5]: I'm not entirely sure, but I think npmignore is not recursive like gitignore is, so this file would have to be moved up to the root dir of this repo

Comment (mycognosist) [reply web/.npmignore]: Yes, good catch. I see that I also had one in the root. I've removed the duplication.

**Review by staltz — APPROVED** [review]
Looks good!

**Pull request merged**